spark-extension/src/main/java/org/apache/spark/sql/auron/JniBridge.java

```diff
@@ -89,10 +90,14 @@ public static boolean isDriverSide() {
     public static FSDataInputWrapper openFileAsDataInputWrapper(FileSystem fs, String path) throws Exception {
         // the path is a URI string, so we need to convert it to a URI object, ref:
         // org.apache.spark.paths.SparkPath.toPath
+        TaskContextHelper$.MODULE$.setThreadNameFromTaskContext();
+        TaskContextHelper$.MODULE$.setHDFSCallerContext();
```

Comment:
HDFS caller context also seems to be a threa local object, maybe we can initialize it in `tokio_runtime_builder.on_thread_start` .